lib/ansible/cli/doc.py

```diff
@@ -328,6 +586,30 @@ def run(self):
                 docs = DocCLI._list_keywords()
             else:
                 docs = DocCLI._get_keywords_docs(context.CLIARGS['args'])
+        elif plugin_type == 'role':
+            if context.CLIARGS['list_dir']:
+                # If an argument was given with --list, it is a collection filter
+                coll_filter = None
+                if len(context.CLIARGS['args']) == 1:
+                    coll_filter = context.CLIARGS['args'][0]
+                    if not AnsibleCollectionRef.is_valid_collection_name(coll_filter):
+                        raise AnsibleError('Invalid collection name (must be of the form namespace.collection): {0}'.format(coll_filter))
+                elif len(context.CLIARGS['args']) > 1:
+                    raise AnsibleOptionsError("Only a single collection filter is supported.")
+
+                list_json = self._create_role_list(roles_path, collection_filter=coll_filter)
+                if do_json:
+                    jdump(list_json)
+                else:
+                    self._display_available_roles(list_json)
+            else:
+                role_json = self._create_role_doc(context.CLIARGS['args'], roles_path,
+                                                  context.CLIARGS['entry_point'])
+                if do_json:
```

Comment:
no need to do this here, we already take care of it below, just update 'docs' variable and 353/635 will display

Reply:
Done